### PR TITLE
fix: fix git url parsing

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Close https://github.com/coder/envbuilder/issues/175

### CODE REVIEW

File: git_test.go [hunk 1]
```diff
@@ -384,6 +384,35 @@ func TestSetupRepoAuth(t *testing.T) {
...
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			url, err := envbuilder.ParseGitURL(tt.gitURL)
+			require.NoError(t, err)
+			require.Equal(t, tt.expected, url.String())
+		})
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply git_test.go:415]: What do you mean by reference?

File: git.go [hunk 2]
```diff
@@ -251,3 +252,8 @@ func SetupRepoAuth(options *Options) transport.AuthMethod {
 	}
 	return auth
 }
+
+// Parses a Git URL and returns a corresponding URL object
+func ParseGitURL(URL string) (*url.URL, error) {
+	return giturls.Parse(URL)
+}
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply git.go:259]: That is true.

File: envbuilder_test.go [hunk 3]
```diff
@@ -9,11 +9,34 @@ import (
...
+			name:     "SSH",
+			gitURL:   "git@github.com:coder/envbuilder.git",
+			expected: "/workspaces/envbuilder",
+		},
+		{
+			name:     "empty",
+			gitURL:   "",
+			expected: envbuilder.EmptyWorkspaceDir,
+		},
+	}
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply envbuilder_test.go:33]: Do you have any examples of these?

File: git_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnstcn) [reply git_test.go:415]: https://github.com/coder/envbuilder/blob/main/README.md#git-branch-selection

> Choose a branch using GIT_URL with a ref/heads reference. For instance:
> 
> GIT_URL=https://github.com/coder/envbuilder-starter-devcontainer/#refs/heads/my-feature-branch

File: envbuilder_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (johnstcn) [reply envbuilder_test.go:33]: https://go.dev/src/net/url/url_test.go#L1604

**Review by johnstcn — APPROVED** [review]
This looks fine to me once we have a couple more test cases for invalid git URLs. I don't think I need to review again.

**Review by johnstcn — APPROVED** [review]

**Pull request merged**